addons/hr_timesheet/views/hr_timesheet_portal_templates.xml

```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <template id="portal_layout" name="Portal layout: timesheet menu entry" inherit_id="portal.portal_breadcrumbs" priority="35">
+        <xpath expr="//ol[hasclass('o_portal_submenu')]" position="inside">
+            <li t-if="page_name == 'timesheet' or timesheet" t-attf-class="breadcrumb-item #{'active' if not timesheet else ''}">
+                <t>Timesheets</t>
+            </li>
+        </xpath>
+    </template>
+
+    <template id="portal_my_home_timesheet" name="Portal My Home : timesheets entries" inherit_id="portal.portal_my_home" priority="45">
+        <xpath expr="//div[hasclass('o_portal_docs')]" position="inside">
+            <t t-if="timesheet_count" t-call="portal.portal_docs_entry">
+                <t t-set="title">Timesheets</t>
+                <t t-set="url" t-value="'/my/timesheets'"/>
+                <t t-set="count" t-value="timesheet_count"/>
+            </t>
+        </xpath>
+    </template>
+
+    <template id="portal_my_timesheets" name="My Timesheets">
+        <t t-call="portal.portal_layout">
+            <t t-set="breadcrumbs_searchbar" t-value="True"/>
+
+            <t t-call="portal.portal_searchbar">
+                <t t-set="title">Timesheets</t>
+            </t>
+            <t t-if="not grouped_timesheets">
+                <div class="alert alert-warning mt8" role="alert">
+                    There are no timesheets.
+                </div>
+            </t>
+            <t t-if="grouped_timesheets">
+                <t t-call="portal.portal_table">
+                    <t t-foreach="grouped_timesheets" t-as="timesheets">
+                        <thead>
```

Comment:
also, can't we have a generic template to render the same timesheet table for task, invoice and timesheet ? 
Field order: Date (display in user lang format), description (with higher than other column), Employee ("responsible" label should be changed), Duration (instead of hours, and displayed with duration widget).

Reply:
@jem-odoo , on the timesheet the view is default group by project_id so UI will be different for that. but for the task and invoice, we can use a generic template. what do you think? Thanks

Reply:
ok, let's do it only for task and invoice then.
don't forget to adapt the other one for the format of dates and unit amount ;)